test/Database/LSMTree/Class/Monoidal.hs

```diff
@@ -12,39 +14,27 @@ module Database.LSMTree.Class.Monoidal (
   , withTableDuplicate
   , withTableMerge
   , withCursor
-      -- * Model 2 instance
-  , runInOpenSession
-  , convLookupResult
-  , convLookupResult'
-  , convQueryResult
-  , convQueryResult'
-  , convUpdate
-  , convUpdate'
-  , MSession (..)
-  , MTableHandle (..)
-  , MCursor (..)
-  , MErr (..)
-  , MBlobRef (..)
+  , module Types
```

Comment:
I'm not a big fan of re-exports, because it makes it hard to follow where identifiers are coming from. However, it's also weird that one would have to import a bunch of other modules. What do you think?

If we change this, we should make sure it's consistent across the model/class modules

I'm currently leaning towards having no re-exports in the `Class` and `Model` modules. The model/class hierarchy is already a little bit complicated, and re-rexports of identifiers might not help

Also, I haven't really been consistent with re-exports vs. no re-exports in these modules myself, so I'm mainly criticising myself :stuck_out_tongue_closed_eyes:

Reply:
I don't think it makes a big difference, but let me share some of my reasoning.

IMHO it's sometimes a bit clearer to have the re-export, for example with `convLookupResult :: Model.LookupResult v b -> Class.LookupResult v b` it makes it clear that it exists because the model's type doesn't quite fit the class we want to implement. If it was something like `-> Real.LookupResult`, the purpose is already less obvious.

Another way of thinking about it: The class uses a lot of the same types as the public API of the implementation. Now the question is: Is that a consciously chosen property of the class? If so, we can expect downstream code to also import something like `LSMTree.Normal`. If it is more of an implementation detail, we should re-export the types and use them as `Class._`, which allows using/implementing the class without considering the implementation, and also makes it easier to swap types out later.